demos/text-generation/llm.js

```diff
@@ -245,13 +245,32 @@ export class LLM {
 
         let attnMask = Array.from({ length: inputIdsLen }, () => BigInt(1));
         const positionIds = Array.from({ length: inputIdsLen }, (_, i) => BigInt(i++));
-        // Padding input_ids, attention_mask, position_ids to have length of this.maxLength
-        const inputIdsBuffer = this.paddingInput(inputIds, this.maxLength);
+        // Both input_ids and position_ids have shapes of [batch_size, sequence_length].
+        // the sequence_length is the length of inputIds, which is dynamic.
+        // Since WebNN does not support dynamic shapes, fix the sequence_length to maxLength and
```

Comment:
> Since WebNN does not support dynamic shapes

Even if WebNN supports dynamic shapes in the future, the reality is that certain backends will need to regenerate the graph whenever an input size changes, and performance-wise, backends execute much faster when known sizes are passed for allocation planning. So it would probably still be worth it to pad them.

(resolve me - no action except to read)